src/main/java/com/hotels/bdp/circustrain/bigquery/util/SchemaExtractor.java

```diff
@@ -40,25 +42,26 @@ public String getSchemaFromStorage(Storage storage, ExtractionUri extractionUri)
     Page<Blob> blobs = storage
         .list(extractionUri.getBucket(), BlobListOption.currentDirectory(),
             BlobListOption.prefix(extractionUri.getFolder() + "/"));
-    Blob file = blobs.iterateAll().iterator().next();
-    return getSchemaFromFile(file);
+    Blob blob = blobs.iterateAll().iterator().next();
+    return extractSchemaFromFile(blob);
   }
 
-  private String getSchemaFromFile(Blob file) {
-    try (SeekableByteArrayInput input = new SeekableByteArrayInput(file.getContent())) {
```

Comment:
The OOM is caused by `file.getContent()` which reads the entire Avro file (which can be hundreds of megabytes) into memory. Instead we need to do some faffing around with the the GCS API to get a channel (it doesn't expose InputStreams directly) and then convert that into an InputStream (as the Avro API accepts these).